tutorials/multi/2_node_classification.py

```diff
@@ -0,0 +1,247 @@
+            for step, (input_nodes, output_nodes, mfgs) in enumerate(tq):
+                # feature copy from CPU to GPU takes place here
+                inputs = mfgs[0].srcdata['feat']
+                labels = mfgs[-1].dstdata['label']
+
+                predictions = model(mfgs, inputs)
+
+                loss = F.cross_entropy(predictions, labels)
+                opt.zero_grad()
+                loss.backward()
+                opt.step()
+
+                accuracy = sklearn.metrics.accuracy_score(labels.cpu().numpy(), predictions.argmax(1).detach().cpu().numpy())
+
+                tq.set_postfix({'loss': '%.03f' % loss.item(), 'acc': '%.03f' % accuracy}, refresh=False)
+
+        model.eval()
+
+        # Evaluate on only the first GPU.
+        if proc_id == 0:
+            predictions = []
+            labels = []
+            with tqdm.tqdm(valid_dataloader) as tq, torch.no_grad():
+                for input_nodes, output_nodes, mfgs in tq:
+                    inputs = mfgs[0].srcdata['feat']
+                    labels.append(mfgs[-1].dstdata['label'].cpu().numpy())
+                    predictions.append(model(mfgs, inputs).argmax(1).cpu().numpy())
+                predictions = np.concatenate(predictions)
+                labels = np.concatenate(labels)
+                accuracy = sklearn.metrics.accuracy_score(labels, predictions)
+                print('Epoch {} Validation Accuracy {}'.format(epoch, accuracy))
+                if best_accuracy < accuracy:
+                    best_accuracy = accuracy
+                    torch.save(model.state_dict(), best_model_path)
+
+        # Note that this tutorial do not train the whole model to the end.
+        break
+
+
+######################################################################
+# Spawning Trainer Processes
+# --------------------------
+# 
+# A typical scenario for multi-GPU training with DDP is to replicate the
+# model once per GPU, and spawn one trainer process per GPU.
+# 
+# PyTorch tutorials recommend using ``multiprocessing.spawn`` to spawn
+# multiple processes. This however is undesirable for training node
+# classification or link prediction models on a single large graph,
+# especially on Linux. The reason is that a single large graph itself may
+# take a lot of memory, and ``mp.spawn`` will duplicate all objects in the
+# program, including the large graph. Consequently, the large graph will
+# be duplicated as many times as the number of GPUs.
+# 
+# To alleviate the problem we recommend using ``multiprocessing.Process``,
+# which *forks* from the main process and allows sharing the same graph
+# object to trainer processes via *copy-on-write*, as long as the graph
+# object is not mutated. This can greatly reduce the memory consumption.
+# 
+# To keep the graph object from mutating, you need to call
+# ``create_formats_`` so that no new sparse matrix representations of the
+# graph will be created on the fly.
+# 
+
+graph.create_formats_()
```

Comment:
Do we have a better name? This `create_formats_` looks internal.

Reply:
The suffix underscore represents an in-place operation like PyTorch.  It's been there long ago and is used in a couple places already.  I would rather keep it like this.

Reply:
I'm not familiar with this API, thus its function here is less intuitive to me: there's a gap between "create new sparse mat rep" and "keep it from mutating". Why is it immutable (is it?) when all sparse formats are created?

Reply:
Elaborated a bit more.  See if this makes sense?

Reply:
Looks much better, thanks! Separately, the topic of formats could be a vignette in user guide, if we don't yet have one.